### DISCUSSION TITLE
How can I get Wi-Fi list and connect Wi-Fi in Avalonia?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#13391] [Q&A] [answered]
I am coding a cross-platform program (Windows and Embed Linux, especially embedded Linux) by 11.0.5.

What I want to achieve is to connect to Wi-Fi in my program. To achieve this, I have to get Wi-Fi list and connect it by code.

Moreover, I hope to be able to dynamically obtain the strength of the Wi-Fi signal like a mobile phone.

I searched on Google, and it seems that only UWP provides related APIs, and there seems to be no related .NET API.

How can I achieve this?

**Comment by kekekeks** [comment, marked as answer]
Linux:
https://people.freedesktop.org/~lkundrak/nm-docs/spec.html
Related XML files to be consumed by Tmds.DBus.Tool or Tmds.DBus.SourceGenerator - https://github.com/NetworkManager/NetworkManager/tree/main/introspection


Windows:
https://learn.microsoft.com/en-us/windows/win32/api/wlanapi/nf-wlanapi-wlanopenhandle
https://learn.microsoft.com/en-us/windows/win32/api/wlanapi/nf-wlanapi-wlanenuminterfaces

**Reply by kyuranger** [reply]
Thanks a lot. Let me try it.
